harness/determined/pytorch/deepspeed/_mpu.py

```diff
@@ -0,0 +1,74 @@
+from typing import cast
+
+from deepspeed.runtime.pipe.topology import PipelineParallelGrid
+
+from determined._core._distributed import DistributedContext
```

Comment:
lol this one I'll give you a pass on because the `_core` module has a temporary name that is in volation of our normal rules for [python code layout](https://github.com/rb-determined-ai/pedl/blob/python-code-layout/python_code_layout.md).

Basically, within Determined we try pretty hard to always have a clear import path for anything.  If there's a symbol in `pkg/_module.py` that you need outside of that file, you expose that symbol in `pkg/__init__.py`, and then you reference it as `pkg.symbol`.

Eventually this should look like:

```python
from determined import core
core.DistributedContext
```

Though for now, it is:
```python
from determined import _core
_core.DistributedContext
```

Basically the name of thefile `_core/_distributed.py` is always going to be an implementation detail, and the symbols exposed by the core API are all in the `_core` namespace (which will become `core` when the Core API becomes official).